runtime/vm/ContinuationHelpers.cpp

```diff
@@ -198,8 +224,19 @@ yieldContinuation(J9VMThread *currentThread)
 	 *
 	 * must be maintained for weakly ordered CPUs, to unsure that once the continuation is again available for GC scan (on potentially remote CPUs), all CPUs see up-to-date stack .
 	 */
-	Assert_VM_true((uintptr_t)currentThread == continuation->state);
-	continuation->state = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+	j9object_t continuationObject = J9VMJAVALANGTHREAD_CONT(currentThread, currentThread->carrierThreadObject);
+	/* Notify GC of Continuation stack swap */
+	jboolean finished = J9VMJDKINTERNALVMCONTINUATION_FINISHED(currentThread, continuationObject);
+
+	if (finished) {
+		VM_VMHelpers::setContinuationFinished(continuation);
+	}
+	Assert_VM_true(VM_VMHelpers::isContinuationMountedWithCarrierThread(continuation->state, currentThread));
+	VM_VMHelpers::resetContinuationCarrierID(continuation);
+	/* calling postUnmountContinuation() compensating scan after resetContinuationCarrierID() to prevent the scan happens before resetContinuationCarrierID()(the state still shows the continuation is mounted). */
```

Comment:
Example of a comment which was difficult to understand.
```suggestion
	/* calling postUnmountContinuation() compensating scan after resetContinuationCarrierID() to prevent the scan happens before resetContinuationCarrierID()(the state still shows the continuation is mounted). */
```

Reply:
@babsingh the base changes has passed sanity.functional on xlinux and local skynet stress test, I just start a new personal build with sanity.openjdk and extended.openjdk on multiple platforms, here is the link (https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/16006/).